From 47e4c1155c8d72dd21cf2ab2df68cf91cce67f7b Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Mon, 14 Jul 2014 13:41:35 -0400
Subject: [x86] mce: Fix mce regression from recent cleanup

Message-id: <1405345303-24468-6-git-send-email-prarit@redhat.com>
Patchwork-id: 85781
O-Subject: [RHEL7.1 PATCH BZ 1119306 05/13] x86/mce: Fix mce regression from recent cleanup
Bugzilla: 1119306
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1119306

commit 1a7f0e3c4fff449f6dd08787beb98a8e57d8cdc7
Author: Tony Luck <tony.luck@intel.com>
Date:   Wed Jul 24 10:09:43 2013 -0700

    x86/mce: Fix mce regression from recent cleanup

    In commit 33d7885b594e169256daef652e8d3527b2298e75
       x86/mce: Update MCE severity condition check

    We simplified the rules to recognise each classification of recoverable
    machine check combining the instruction and data fetch rules into a
    single entry based on clarifications in the June 2013 SDM that all
    recoverable events would be reported on the unaffected processor with
    MCG_STATUS.EIPV=0 and MCG_STATUS.RIPV=1.  Unfortunately the simplified
    rule has a couple of bugs.  Fix them here.

    Acked-by: Naveen N. Rao <naveen.n.rao@linux.vnet.ibm.com>
    Signed-off-by: Tony Luck <tony.luck@intel.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kernel/cpu/mcheck/mce-severity.c b/arch/x86/kernel/cpu/mcheck/mce-severity.c
index e270352..c370e1c 100644
--- a/arch/x86/kernel/cpu/mcheck/mce-severity.c
+++ b/arch/x86/kernel/cpu/mcheck/mce-severity.c
@@ -111,8 +111,8 @@ static struct severity {
 #ifdef CONFIG_MEMORY_FAILURE
  MCESEV(
   KEEP, "Action required but unaffected thread is continuable",
-  SER, MASK(MCI_STATUS_OVER|MCI_UC_SAR|MCI_ADDR|MCACOD, MCI_UC_SAR|MCI_ADDR),
-  MCGMASK(MCG_STATUS_RIPV, MCG_STATUS_RIPV)
+  SER, MASK(MCI_STATUS_OVER|MCI_UC_SAR|MCI_ADDR, MCI_UC_SAR|MCI_ADDR),
+  MCGMASK(MCG_STATUS_RIPV|MCG_STATUS_EIPV, MCG_STATUS_RIPV)
   ),
  MCESEV(
   AR, "Action required: data load error in a user process",
-- 
1.7.1